### ISSUE TITLE
use treebank data from Arethusa for disambiguation of popup

### ISSUE THREAD

**Issue by balmas** [alpheios-project/alpheios-core#146]
Currently Alpheios can retrieve treebank data for disambiguating the information displayed in the Alpheios popup  (i.e. lemma and inflection) but it retrieves it from a separate location than the trees that are displayed in the diagram panel using Arethusa. For this feature as it currently functions, the trees need to be loaded in an Alpheios-hosted eXist server. 

Change this so that the same environment that is used to display the tree in the diagram window (e.g. using the Perseids treebank GitHub template) can be used to serve the data for disambiguating the popup data. 

This item is dependent upon corresponding changes to the Arethusa code and to the Perseids treebank template.

**Comment by balmas** [comment]
The prototype design

![arethusa_messaging_client](https://user-images.githubusercontent.com/644870/75264897-1a029100-57be-11ea-95a5-9f0eca0db587.png)

**Comment by balmas** [comment]
The above prototype design is backwards-compatible, so that the existing Treebank behavior will continue to work. New behavior is triggered by a `data-alpheios_tb_version` property on the page element which defines the source of the treebank (via the `data-alpheios_tb_src`) attribute.  If the version is defined and >= 3 then the new arethusaAdapter is used, otherwise the old tbAdapter is used.

e.g. the following page data on a page with the embedded library would trigger the new adapter:

```
  <meta name="alpheios-v3-treebank-diagram-url"
      data-alpheios_tb_version="3"
      data-alpheios_tb_src="https://alpheios-project.github.io/treebank-template/embed/DOC/SENTENCE?w=WORD"/>
   <div lang="grc" class="alpheios-enabled">
     <span data-alpheios_tb_ref="on-the-murder-of-eratosthenes-1-50#1-1">περὶ</span>
     <span data-alpheios_tb_ref="on-the-murder-of-eratosthenes-1-50#1-2">πολλοῦ</span>
     <span data-alpheios_tb_ref="on-the-murder-of-eratosthenes-1-50#1-3">ἂν</span>
     <span data-alpheios_tb_ref="on-the-murder-of-eratosthenes-1-50#1-4">ποιησαίμην</span>
     <span data-alpheios_tb_ref="on-the-murder-of-eratosthenes-1-50#2-1">εὖ</span>
     </span>
     <span data-alpheios_tb_ref="on-the-murder-of-eratosthenes-1-50#2-2">γὰρ</span>
     <span data-alpheios_tb_ref="on-the-murder-of-eratosthenes-1-50#2-3">οἶδʼ</span>
     <span data-alpheios_tb_ref="on-the-murder-of-eratosthenes-1-50#2-4">ὅτι</span>
   </div>
```

**Comment by balmas** [comment]
There are the following issues with the prototype:

1) There is a known problem with Arethusa, that if the tree isn't visible when it is first rendered, the display will be messed up. In the prior solution, we just waited to set the source url of the iframe until the panel was actually opened.  We cannot do that in the new solution, because we need the iframe to be loaded before we issue our first morphology request of it and that is before the panel is opened to display the tree.  I added a `refreshView` operation to the arethusa API to address this, which can be used to refresh the tree display when it is displayed.  I have for now hacked use of this into the `uiController.changeTab` method -- having it issue a refreshView request message, but it probably doesn't belong there.  

2) There is also a separate issue with the tree display when the panel is closed and reopened in between displays of different words. The display gets only partially drawn until the window is resized in some way (e.g. if you just open the developer console, the tree suddenly appears). I think this is caused by Vue -- I cannot reproduce it in a page that doesn't use Vue. I also confirmed that if the panel does not close in between displays of the tree, then the problem doesn't occur.   I suspect use of nextTick  might help here although I'm not entirely sure where to put it.  (See more on this below)

3) Right now, we require each word that is linked to a treebank to have both the sentence identifier and word identifier defined for it.  We also need to support linking at the sentence level -- as described in #148 -- the design should eventually accommodate this.  It will require changes to Arethusa too - as described at alpheios-project/arethusa#791  

I think the final design probably needs a module to more finely control the integration between the the Alpheios and Arethusa state.  The management of the single iframe that is housed in the Vue component but used by the Client Adapter needs to be more reliable that it is now. We should probably also use the new Arethusa api to manage the navigation of the document within Arethusa entirely, rather that resetting the source url of the frame each time the user selects a new word. That might also help with the redrawing problem described above.

**Comment by balmas** [comment]
@kirlat I'd like to ask you to review what I have done so far, and then see if you can take this design from here. I could do a PR for the prototype or we could leave it in the branch for you to take over.

**Comment by balmas** [comment]
@kirlat I'd like to ask you to review what I have done so far, and then see if you can take this design from here. I could do a PR for the prototype or we could leave it in the branch for you to take over.

**Comment by kirlat** [comment]
I'm wondering why do we do any word-related requests at the early stage of a page load, when no word seems to be selected:
![image](https://user-images.githubusercontent.com/18631055/76007598-bbf54c80-5f27-11ea-88e1-d490688a5637.png)
Is that because of some specifics of an architecture?

**Comment by balmas** [comment]
> I'm wondering why do we do any word-related requests at the early stage of a page load, when no word seems to be selected:
> ![image](https://user-images.githubusercontent.com/18631055/76007598-bbf54c80-5f27-11ea-88e1-d490688a5637.png)
> Is that because of some specifics of an architecture?

we need to make sure the iframe is loaded before the first word request happens.  Right now, the way the state is setup, the iframe only gets updated if we have a word specified. I think this is one of the things we have to improve upon.

**Comment by balmas** [comment]
Either way, we're going to need to load the sentence a word is in before we issue a request for morphology for that word but the timing of when and how we load the first sentence is what could probably be improved upon.

**Comment by balmas** [comment]
the basics of this are implemented via #219. Additional issues will be added for remaining bugs to be worked out.

**Comment by balmas** [comment]
in Alpheios Components 3.3.0-qa.20200410462 but wait on #221 before testing

**Comment by balmas** [comment]
assigning to myself for pre-testing before passing to @monzug

**Comment by balmas** [comment]
ok, ready for testing in Alpheios Components 3.3.0-qa.20200504341

As with  #166 - test by looking at a variety of words in treebanked texts which should be represented as disambiguated and questioning me on any that aren't.

**Comment by monzug** [comment]
Bridget, I haven't found any word without the red arrow for disambiguated. do you have any to suggest?

**Comment by balmas** [comment]
well that's good then :-) 

The use cases for disambiguation are described at https://github.com/alpheios-project/alpheios-core/issues/166#issuecomment-607300619 so if you think you've covered most of those then I would say this can be verified.

**Comment by monzug** [comment]
I verified this one because I was not able to find any words not disambiguated
